crates/editor/src/editor.rs

```diff
@@ -2564,14 +2564,47 @@ impl Editor {
             }
         }
 
-        self.change_selections(auto_scroll.then(|| Autoscroll::newest()), cx, |s| {
-            if !add {
-                s.clear_disjoint();
-            } else if click_count > 1 {
-                s.delete(newest_selection.id)
+        let point_to_delete: Option<usize> = {
+            let selected_points: Vec<Selection<Point>> =
+                self.selections.disjoint_in_range(start..end, cx);
+
+            if !add || click_count > 1 {
+                None
+            } else if selected_points.len() > 0 {
+                Some(selected_points[0].id)
```

Comment:
What exactly does this try to do?

Reply:
This piece is for the selected range, if I double click on a word with multicursor and I try to remove the selection I need to get the selection ID. You can try by placing some cursors, double click a word and then click on the selected word with the double_cursor_keyword pressed and you will see that the selection is removed correctly

Reply:
This because when my click is inside the selection, by taking the selections between a point I expect to have at most one selection or 0, and if I have any I delete it @mrnugget

Reply:
Got it, makes sense!